app/src/main/java/com/github/orkest/View/MainActivity2.kt

```diff
@@ -0,0 +1,70 @@
+package com.github.orkest.View
+
+
+import android.os.Bundle
+import android.util.Log
+import androidx.appcompat.app.AppCompatActivity
+import com.github.orkest.R
+import com.spotify.android.appremote.api.ConnectionParams
+import com.spotify.android.appremote.api.Connector.ConnectionListener
+import com.spotify.android.appremote.api.SpotifyAppRemote
+import com.spotify.protocol.types.PlayerState
+
+
+class MainActivity2 : AppCompatActivity() {
+    private var mSpotifyAppRemote: SpotifyAppRemote? = null
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main2)
+    }
+
+    override fun onStart() {
+        super.onStart()
+        val connectionParams = ConnectionParams.Builder(CLIENT_ID)
+            .setRedirectUri(REDIRECT_URI)
+            .showAuthView(true)
+            .build()
+
+        SpotifyAppRemote.connect(this, connectionParams,
+            object : ConnectionListener {
+                override fun onConnected(spotifyAppRemote: SpotifyAppRemote) {
+                    mSpotifyAppRemote = spotifyAppRemote
+                    Log.d("MainActivity", "Connected! Yay!")
+
+                    // Now you can start interacting with App Remote
+                    connected()
+                }
+
+                override fun onFailure(throwable: Throwable) {
+                    Log.e("MyActivity", throwable.message, throwable)
+
+                    // Something went wrong when attempting to connect! Handle errors here
+                }
+            })
+    }
+
+    override fun onStop() {
+        super.onStop()
+        SpotifyAppRemote.disconnect(mSpotifyAppRemote)
+    }
+
+    private fun connected() {
+        // Play a playlist
+        mSpotifyAppRemote!!.playerApi.play("spotify:playlist:37i9dQZF1DX2sUQwD7tbmL")
+
+        // Subscribe to PlayerState
+        mSpotifyAppRemote!!.playerApi
+            .subscribeToPlayerState()
+            .setEventCallback { playerState: PlayerState ->
+                val track = playerState.track
+                if (track != null) {
+                    Log.d("MainActivity", track.name + " by " + track.artist.name)
+                }
+            }
+    }
+
+    companion object {
+        private const val CLIENT_ID = "e7ac920406d54975bc79962dec94f4ab"
+        private const val REDIRECT_URI = "https://com.github.orkest/callback/"
+    }
```

Comment:
Nice that you added a companion object and not static keywords to respect the language design